hack/generator/pkg/codegen/golden_files_test.go

```diff
@@ -150,7 +150,7 @@ func NewTestCodeGenerator(testName string, path string, t *testing.T, testConfig
 	case config.GenerationPipelineAzure:
 		codegen.RemoveStages("deleteGenerated", "rogueCheck", "createStorage", "reportTypesAndVersions")
 		if !testConfig.HasArmResources {
-			codegen.RemoveStages("createArmTypes", "applyArmConversionInterface")
+			codegen.RemoveStages("createArmTypes", "applyArmConversionInterface", "removeEmbeddedResources", "removeEmptyObjectTypes")
```

Comment:
This is a great question. The problem comes from the approach that the `removeEmbeddedResources` stage takes. That stage is looking at things not as a collection of types but as a set of graphs of types with the roots of the graph being the resources. The problem is that, if there _are_ no resources in the collection of types than this stage ends up behaving the same as the `stripUnreferenced` phase - it removes all types that do not root up to a resource.

When you _have_ types rooting to resources that's fine but when there are none everything is removed and nothing is generated.

Technically speaking we could make the stage handle this case by special casing it, but since it only really makes sense in tests it seemed better to remove them here.

Reply:
I've left a comment here explaining that.